dev_guide/configmaps.adoc

```diff
@@ -0,0 +1,285 @@
+= ConfigMaps
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+
+The `ConfigMap` object type provides a mechanism to hold non-sensitive
+configuration information such as whole configuration files, property values,
+etc. ConfigMaps separate configuration data from the pods that use it and can
+be mounted into containers using a volume plug-in or injected as an
+environment variable.  This topic discusses important properties of ConfigMaps
+and provides an overview on how developers can use them.
+
+ConfigMaps is similar to the link:secrets.html[Secrets] type, but is
+designed for use with non-sensitive information.  For sensitive information,
+use Secret instead of ConfigMap. 
+
+====
+----
+{
+  "apiVersion": "v1",
+  "kind": "ConfigMap",
+  "metadata": {
+    "name": "example-configmap"
+  },
+  "namespace": "my-namespace",
+  "data": { <1>
+    "should-work-right": "true",
+    "sometimes-be-confusing": "false"
+    "whole-files-supported": "supported=why yes they are\nhow-enthusiastically=gladly"
+    }
+}
+----
+<1> The allowable format for the keys in the `data` field must meet the
+guidelines in the *DNS_SUBDOMAIN* value in
+https://github.com/GoogleCloudPlatform/kubernetes/blob/v1.0.0/docs/design/identifiers.md[the
```

Comment:
old link, copied from secrets (it's still that way in the secrets docs -- how about i fix them together after this merges.